pyuvdata/uvdata.py

```diff
@@ -1774,3 +1775,147 @@ def antpairpol_iter(self, squeeze='default'):
         antpairpols = self.get_antpairpols()
         for key in antpairpols:
             yield (key, self.get_data(key, squeeze=squeeze))
+
+
+    def parse_ants(self,ant_str):
+        """
+        Generates two lists of antenna pair tuples and polarization indices based on
+        parsing of the string ant_str.  If no valid polarizations or antenna numbers
+        are found in ant_str (Stokes params, or combinations of [lr] or [xy]),
+        ant_pairs_nums and polarizations are returned as None.
+
+        Args:
+            ant_str: String containing antenna information to pass to select function.
+                Can be all, auto, cross, or combinations of antenna numers and polarization
+                indicators l and r or x and y.
+
+        Output:
+            ant_pairs_nums: List of tuples containing the parsed pairs of antennae numbers.
+                If 'all' is passed as ant_str, returned as None.
+            polarizations: List of desired polarizations.  If no polarizations found in ant_str
+                then returned as None.
+        """
+
+        ant_re = r'(\(((-?\d+[lrxy]?,?)+)\)|-?\d+[lrxy]?)'
+        bl_re = '(^(%s_%s|%s),?)' % (ant_re, ant_re, ant_re)
```

Comment:
This came straight from `aipy/scripting.py` but I did add support for circular polarizations.